go/cmd/server/server.go

```diff
@@ -28,6 +30,53 @@ func (s *servingServiceServer) GetOnlineFeatures(ctx context.Context, request *s
 	if err != nil {
 		return onlineResponse, err
 	}
-	go flushToChannel(s, onlineResponse, request)
+	//Synchronous since adding logs to channel is a fast operation.
+	generateLogs(s, onlineResponse, request)
 	return onlineResponse, err
 }
+
+func generateLogs(s *servingServiceServer, onlineResponse *serving.GetOnlineFeaturesResponse, request *serving.GetOnlineFeaturesRequest) {
+	for _, featureVector := range onlineResponse.Results {
+		featureValues := make([]*types.Value, len(featureVector.Values))
+		eventTimestamps := make([]*timestamppb.Timestamp, len(featureVector.EventTimestamps))
+		for idx, featureValue := range featureVector.Values {
+			if featureVector.Statuses[idx] != serving.FieldStatus_PRESENT {
+				continue
+			}
+			featureValues[idx] = &types.Value{Val: featureValue.Val}
+		}
+		for idx, ts := range featureVector.EventTimestamps {
+			if featureVector.Statuses[idx] != serving.FieldStatus_PRESENT {
+				continue
+			}
+			eventTimestamps[idx] = &timestamppb.Timestamp{Seconds: ts.Seconds, Nanos: ts.Nanos}
+		}
+		// TODO(kevjumba): For filtering out and extracting entity names when the bug with join keys is fixed.
```

Comment:
You can't commit this code into master. Please remove it or at change this PR into a WIP or draft so that people don't spend time reviewing it.